lib/chef/http/basic_client.rb

```diff
@@ -139,7 +139,8 @@ def env
 
       def http_client_builder
         http_proxy = proxy_uri
-        if http_proxy.nil?
+        if !http_proxy.respond_to?(:host) || !http_proxy.respond_to?(:port) ||
+           !http_proxy.respond_to?(:user) || !http_proxy.respond_to?(:password)
```

Comment:
`if !http_proxy.is_a?(URI)`

or for bonus points drop the '!' and invert the if-else clauses

the #proxy_uri method also looks like it should be documented that it returns either a URI object or it looks like...  false?  and tests added around that if they have to be...